lib/collabland/__tests__/disconnectSpace.spec.ts

```
@@ -0,0 +1,113 @@
+import { DataNotFoundError } from '@charmverse/core/errors';
+import { prisma } from '@charmverse/core/prisma-client';
+import { testUtilsUser } from '@charmverse/core/test';
+import { v4 } from 'uuid';
+
+import { createDiscordUser } from 'testing/utils/discord';
+import { addUserToSpace } from 'testing/utils/spaces';
+
+import { disconnectSpace } from '../disconnectSpace';
+
+describe('disconnectSpace', () => {
+  it('should throw error if no user with the provided discord id exist', async () => {
```

i think it would be better if the function did not return an error in this case. It forces the code to deal with this specific error, because it's not really a problem. The action is already fulfilled, so the code that's calling it shouldn't need to worry then. If it's useful, you could return an error object like "{ notFound: true }"

---

Good point. Updated to not throw an error and just return `{spaceIds: []}` indicating no spaces where updated, which would be the same thing if a non admin user triggered this event.